proxy/http/remap/NextHopConsistentHash.cc

```diff
@@ -326,6 +326,7 @@ NextHopConsistentHash::findNextHop(TSHttpTxn txnp, void *ih, time_t now)
         // check if the host is retryable.  It's retryable if the retry window has elapsed
         _now == 0 ? _now = time(nullptr) : _now = now;
         if ((pRec->failedAt.load() + retry_time) < static_cast<unsigned>(_now)) {
+          NH_Debug(NH_DEBUG_TAG, "[%" PRIu64 "] next hop %s, retriers: %d", sm_id, pRec->hostname.c_str(), pRec->retriers.load());
```

Comment:
For example suppose retriers == max_retriers - 1.  Then:
1.  one core executes line 330, making retriers == max_retriers, so it goes to the else clause.
2. another core executes line 330, making retriers == max_retriers + 1, so it also goes to the else clause.
Both cores will then execute line 341, making retriers == max_retriers - 1 again.  But neither one acted as a retrier.

Reply:
I think that was the idea with the else clause @ywkaras, to negate the increment on line 330 if they aren't retriers.  Am I missing something here?

Reply:
With that said, I like how you've abstracted the retriers in the other PR.

Reply:
That was the idea of the else clause, to back out the increment if the transaction on that core wasn't going to be a retrier.  Am I missing something @ywkaras?  I do like how you've abstracted this in the other PR though.

Reply:
Is there some way you guarantee the thread interleaving scenario I describe above can not happen?